modules/core/shared/src/main/scala/eu/timepit/refined/types/string.scala

```diff
@@ -17,6 +17,11 @@ object string {
   type TrimmedString = String Refined MatchesRegex[W.`"""^(?!\\s).*(?<!\\s)"""`.T]
 
   object TrimmedString extends RefinedTypeOps[TrimmedString, String]
+
+  /** A `String` representing a hexadecimal number */
+  type HexStringSpec = MatchesRegex[W.`"""([0-9a-f]+)|([0-9A-F]+)"""`.T]
```

Comment:
It seems that this does not work with Scala.js (there is a test failure), but fortunately you've [experience with such problems](https://github.com/fthomas/refined/pull/357#issuecomment-359003701). :smiley_cat:

Reply:
Oh god... I shall stop using `coreJVM/test` and dive into regex' in JS once again :D